### PR TITLE
chore: move hash inside user operation

### PR DESCRIPTION
**Pull request by Vid201**
This PR refactors user operation structs into three structs:
- UserOperation (hash + canonical user operation)
- UserOperationSigned (canonical user operation)
- UserOperationRequest (user operation with some optional fields, used for RPC estimateUserOperationGas)

### CODE REVIEW

File: crates/grpc/src/protos/types/types.proto [hunk 1]
```diff
@@ -26,6 +26,21 @@ message SupportedEntryPoint {
 }
 
 message UserOperation {
+    types.H256 hash = 1;
+    types.H160 sender = 2;
```
Comment (zsluedem) [inline crates/grpc/src/protos/types/types.proto:30]: I am curious why the rest is not using `UserOperationSigned` .

**Review by zsluedem — COMMENTED** [review]
Left a question

File: crates/grpc/src/protos/types/types.proto (hunk 1 above)
Comment (Vid201) [reply crates/grpc/src/protos/types/types.proto:30]: Not sure what exactly you mean?

Comment (zsluedem) [reply crates/grpc/src/protos/types/types.proto:30]: I meant somthing like 
```
message UserOperation {
    types.H256 hash = 1;
    UserOperationSigned uo = 2;
}
```

**Comment by Vid201** [comment]
@zsluedem Addressed the comment

**Review by zsluedem — APPROVED** [review]
LGTM

**Pull request merged**